pkg/rpcclient/actor/maker.go

```diff
@@ -0,0 +1,195 @@
+package actor
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/neorpc/result"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/vmstate"
+)
+
+// TransactionCheckerModifier is a callback that receives the result of
+// test-invocation and the transaction that can perform the same invocation
+// on chain. This callback is accepted by methods that create transactions, it
+// can examine both arguments and return an error if there is anything wrong
+// there which will abort the creation process. Notice that when used this
+// callback is completely responsible for invocation result checking, including
+// checking for HALT execution state (so if you don't check for it in a callback
+// you can send a transaction that is known to end up in FAULT state). It can
+// also modify the transaction (see TransactionModifier).
+type TransactionCheckerModifier func(r *result.Invoke, t *transaction.Transaction) error
+
+// TransactionModifier is a callback that receives the transaction before
+// it's signed from a method that creates signed transactions. It can check
+// fees and other fields of the transaction and return an error if there is
+// anything wrong there which will abort the creation process. It also can modify
+// Nonce, SystemFee, NetworkFee and ValidUntilBlock values taking full
+// responsibility on the effects of these modifications (smaller fee values, too
+// low or too high ValidUntilBlock or bad Nonce can render transaction invalid).
+// Modifying other fields is not supported. Mostly it's useful for increasing
+// fee values since by default they're just enough for transaction to be
+// successfully accepted and executed.
+type TransactionModifier func(t *transaction.Transaction) error
```

Comment:
Maybe it's worth to mention here that once called, transaction hash and size are cached? Just for the record for external users.

Reply:
I'm not sure this adds any value. These hooks are used in methods that sign the transaction, and to perform signing it'll be hashed anyway making any subsequent modifications impossible. So if a user calls `Hash()` in this hook it doesn't change much.